core/src/main/java/org/radargun/stages/test/legacy/AbstractCompletion.java

```diff
@@ -0,0 +1,39 @@
+package org.radargun.stages.test.legacy;
+
+import org.radargun.logging.Log;
+import org.radargun.logging.LogFactory;
+import org.radargun.utils.TimeService;
+
+/**
+ * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+ */
+public abstract class AbstractCompletion implements Completion {
+   protected final static String PROGRESS_STRING = "Number of operations executed by this thread: %d. Elapsed time: %s. Remaining: %s. Total: %s.";
+   protected final static Log log = LogFactory.getLog(Completion.class);
+
+   protected boolean started, completed;
+   protected long startTime;
+   private Runnable handler;
+
+   @Override
+   public synchronized void start() {
+      if (!started) {
+         startTime = TimeService.nanoTime();
+         started = true;
+      }
+   }
+
+   @Override
+   public synchronized void setCompletionHandler(Runnable handler) {
```

Comment:
I think that lambda expression will be used for handling this now as we are based on Java 8 now, so introducing special interface would not have the effect. I'll fix the usage.